eosclubhouse/portal.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (C) 2020 Endless OS Foundation LLC.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
+import logging
+
+from gi.repository import Gio, GLib
+
+
+_logger = logging.getLogger(__name__)
+
+
+class Portal:
+    _proxy = None
+
+    @classmethod
+    def _get_proxy(klass):
+        if klass._proxy is None:
+            klass._proxy = Gio.DBusProxy.new_for_bus_sync(
+                Gio.BusType.SESSION,
+                0,
+                None,
+                'org.freedesktop.portal.Desktop',
+                '/org/freedesktop/portal/desktop',
+                'org.freedesktop.portal.Background',
+                None
+            )
+        return klass._proxy
+
+    @classmethod
+    def request_background(klass, message=''):
+        try:
+            proxy = klass._get_proxy()
+            options = {}
+            if message:
+                options['reason'] = GLib.Variant('s', message)
+            proxy.RequestBackground('(sa{sv})', '', options)
```

Comment:
Thanks for this long and detailed description of the service, I'll take a second look to try to improve the integration. I've been doing some testing with a fedora 33 virtual machine and it looks like that just with the request, the inactivity dialog dissapears, but maybe that's just a implementation issue and this can fail in other distributions, so I'll try to implement the whole flow.

Reply:
I expect what might be happening there is it gets _denied_ permission, rather than simply not granted. (Which feels like a bug in the portal, come to think of it).

So then it is technically not allowed to run in the background, although I think what happens from there is undefined.

Reply:
I've detected that if the user allows, the app will run in background without any problem, and if the user declines, the app will be stopped when the window disappears, after a few seconds. But there's no notification, if the app request the background permission, the system do that automatically without asking again or notify to the user.